diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/SingleLineStatementsRule.kt

```diff
@@ -0,0 +1,87 @@
+package org.cqfn.diktat.ruleset.rules
+
+import com.pinterest.ktlint.core.KtLint
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.SEMICOLON
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.ruleset.constants.Warnings.MORE_THAN_ONE_STATEMENT_PER_LINE
+import org.cqfn.diktat.ruleset.utils.*
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.PsiWhiteSpaceImpl
+import org.jetbrains.kotlin.com.intellij.psi.tree.TokenSet
+
+class SingleLineStatementsRule : Rule("statement") {
+
+    companion object {
+        val semicolonToken = TokenSet.create(SEMICOLON)
+        const val minListTextSize = 1
+    }
+
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var fileName: String? = null
+    private var isFixMode: Boolean = false
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        fileName = params.fileName
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        checkSemicolon(node)
+    }
+
+    private fun checkSemicolon(node: ASTNode) {
+        node.getChildren(semicolonToken).forEach {
+            if (!isSemicolonInMidLine(it)) {
+                if (isError(it)) {
+                    MORE_THAN_ONE_STATEMENT_PER_LINE.warnAndFix(configRules, emitWarn, isFixMode, findWrongText(it),
+                            it.startOffset) {
+                        node.addChild(PsiWhiteSpaceImpl("\n"), it)
+                        node.removeChild(it)
+                    }
+                }
+            }
+        }
+    }
+
+    private fun isError (node: ASTNode) = (node.treeNext != null && !node.isFollowedByNewline())
+            || (node.treeParent.treeNext != null && !node.treeParent.isFollowedByNewline())
+
+    private fun isSemicolonInMidLine(node: ASTNode) = (node.treeNext != null && node.isFollowedByNewline())
```

Comment:
Doesn't it actually check that semicolon is at end of line? Like ```val a = 0;\n``` will return true.